src/platform/ESP32/CHIPMem-Platform.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements heap memory allocation APIs for CHIP. These functions are platform
+ *      specific and might be C Standard Library heap functions re-direction in most of cases.
+ *
+ */
+
+#include <lib/support/CHIPMem.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <esp_heap_caps.h>
+#include <stdlib.h>
+
+namespace chip {
+namespace Platform {
+
+CHIP_ERROR MemoryAllocatorInit(void * buf, size_t bufSize)
+{
+    return CHIP_NO_ERROR;
+}
+
+void MemoryAllocatorShutdown() {}
+
+void * MemoryAlloc(size_t size)
+{
+#ifdef CONFIG_CHIP_MEM_ALLOC_MODE_INTERNAL
```

Comment:
I would suggest to split into 3 separate files and control their inclusion at build times, so instead of ifdefs we would have: `CHIPMem-Platform-interal.cpp`, `CHIPMem-Platform-spiram.cpp` and such. That makes it clearer what calls are paired in what ways.

Can the build system do that?

We also try to not use `ifdef` or `if defined` throughout the source code because sometimes the include paths are complex enough that we are not getting the includes we expect and mismatches are hard to detect. Using separate files would make things easier since both duplicate include or missing include would be found at link time.

Reply:
Done